src/box/xrow.c

```diff
@@ -1705,70 +1705,79 @@ xrow_decode_ballot(const struct xrow_header *row, struct ballot *ballot)
 	return -1;
 }
 
-int
-xrow_encode_register(struct xrow_header *row,
-		     const struct tt_uuid *instance_uuid,
-		     const struct vclock *vclock)
-{
-	memset(row, 0, sizeof(*row));
-	size_t size = mp_sizeof_map(2) +
-		      mp_sizeof_uint(IPROTO_INSTANCE_UUID) +
-		      mp_sizeof_str(UUID_STR_LEN) +
-		      mp_sizeof_uint(IPROTO_VCLOCK) +
-		      mp_sizeof_vclock_ignore0(vclock);
-	char *buf = (char *) region_alloc(&fiber()->gc, size);
-	if (buf == NULL) {
-		diag_set(OutOfMemory, size, "region_alloc", "buf");
-		return -1;
-	}
-	char *data = buf;
-	data = mp_encode_map(data, 2);
-	data = mp_encode_uint(data, IPROTO_INSTANCE_UUID);
-	data = xrow_encode_uuid(data, instance_uuid);
-	data = mp_encode_uint(data, IPROTO_VCLOCK);
-	data = mp_encode_vclock_ignore0(data, vclock);
-	assert(data <= buf + size);
-	row->body[0].iov_base = buf;
-	row->body[0].iov_len = (data - buf);
-	row->bodycnt = 1;
-	row->type = IPROTO_REGISTER;
-	return 0;
-}
+/**
+ * A template which can represent any replication request - join, register, ack,
```

Comment:
I'd rather not mention acks here. You create a separate struct for them right in the next commit.

Reply:
Removed.